### ISSUE TITLE
Add optional transcriptome/proteome clustering step using CD-HIT

### ISSUE THREAD

**Issue by titodamiani** [transXpress/transXpress#82]
Add option to cluster redundant sequences in the assembled proteome using [CD-HIT](https://sites.google.com/view/cd-hit). E.g., `cd-hit -c 1.00` clusters 100% identical sequences, which are likely assembly artefacts. As a next step, headers from the clustered proteome `transcriptome_clstr.pep` are used to fish out corresponding sequences in `transcriptome.fasta` and create a filtered transcriptome `transcriptome_clstr.fasta`. Kallisto is then run on `transcriptome_clstr.fasta` so that raw-reads are aligned on a reduced number of transcripts. See new workflow in the attached image. @komorn is working on this. 

**To-do**:
- Add `cd-hit = TRUE/FALSE` option in the `config.yaml` file. CD-HIT parameters can be set in the Snakefile
- CD-HIT is applied on the `transcriptome.pep` file, which is overwritten with the clustered file. @tomas-pluskal @photocyte do you have anything against this and want to keep both files as `transcriptome.pep` and `transcriptome_clstr.pep`? The first option will make things simpler.
- @komorn I would use the following options as default in the Snakefile: `cd-hit -i transcriptome.pep -o transcriptome_clstr.pep -c 1.00 -n 5 -g 1 -d 0`
- CD-HIT outputs two files (see example files [here](https://drive.google.com/drive/folders/10kz23VnqdC9Twjt-Yv1RJPN4e614bh77?usp=sharing)):
     - `output.pep`: clustered proteome with very same format as the input file. Nothing should break downstream
     - `output.pep.clstr`: file containing the information about the clustered sequences. @komorn already has a Python script for converting this into TSV. @komorn I think we decided to keep the columns as in the `clusters.tsv` example file [here](https://drive.google.com/drive/folders/10kz23VnqdC9Twjt-Yv1RJPN4e614bh77?usp=sharing).    
- Headers from `transcriptome_clstr.pep` are used to fish out the corresponding transcripts in the `transcriptome.fasta` file and create `transcriptome_clstr.fasta`. @komorn already has a Python script for this.

![tXp_new](https://github.com/transXpress/transXpress/assets/65198484/5a15ecd7-d71b-46fc-81d4-97d4933ded8f)

**Comment by komorn** [comment]
Hi @titodamiani, we agreed to only run kallisto once which means we overwrite `transcriptome.pep` if cd-hit is set to true and we just copy the input to the output file in the cd-hit rule if cd-hit option is set to false.

**Comment by titodamiani** [comment]
I agree. It will make things simpler. So both the `transcriptome.pep` and `transcriptome.fasta` are overwritten with the clustered files and then the pipeline proceed as usual.

**Comment by komorn** [comment]
These changes have been implemented. However, by default the pipeline does not produce the `clusters.tsv` file. To maintain flexibility, a note has been added to the config.yaml file: users can set the cd-hit parameter to true (false by default) which will allow them the add `clusters.tsv` file to the list of files in rule all in the Snakefile.

For now all the original and clustered files (e.g. `transcriptome.fasta`, `transcriptome.pep`, `transcriptome_clst.pep`, `transcriptome_clst.fasta`) remain in the main (transXpress) directory. The clustered files are now named differently, but are still present along with the non-clustered ones. @titodamiani proposed to create a cd-hit folder and move the original proteome and transcriptome files to it, leaving only the clustered files in the main transXpress folder. This modification has not yet been implemented, but it can be done by simple edits.
 
Thus, if running cd-hit, we would have:
- the clustered files: `transcriptome.pep`  and `transcriptome.fasta` in the main folder 
- the original files (also named) `transcriptome.pep`  and `transcriptome.fasta` in the cd-hit folder.

This way the user would be able to look at the differences afterwards. Currently all files are in the main folder and from cd-hit onwards the file `transcriptome_clst.pep` is used  (not `transcriptome.pep` as when cd-hit was not implemented yet).

I think it still make sense to name the clustered files differently when the cd-hit option is set to true because it will be easier to distinguish which file is which.

Also, @titodamiani proposed not to name the fasta file created by fishing out the corresponding transcripts from `transcriptome_clstr.pep` in the `transcriptome.fasta` file `transcriptome_clst.fasta`, but `transcriptome_filtered.fasta`

**Comment by titodamiani** [comment]
@komorn Summary:
- Create `cd-hit` folder, which will contain:
   - `transcriptome.fasta` and `transcriptome.pep` (original files)
   - `cluter.tsv`
- Use `transcriptome_filt.fasta` instead of `transcriptome_clstr.fasta`

So in the main transXpress folder there will be `transcriptome_filt.fasta` and `transcriptome_clstr.pep` as non-annotated files and `transcriptome_annoated.fasta` and `transcriptome_annotated.pep` ad annotated files.

Note: annotations (Rfam, Pfam, etc.) are run on the original (non-clustered) files, which means waste of computational resources. This will be fixed in the future. 

Thank you @komorn!!
